include/seqan3/alignment/matrix/alignment_optimum.hpp

```diff
@@ -14,74 +14,119 @@
 {
-    //!\brief The optimal score.
+    //!\brief The index type used to store the alignment coordinates of the optimum.
+    using index_t = IMPLEMENTATION_DEFINED;
+
+    //!\brief The index of the alignment matrix column.
+    index_t column_index{};
+    //!\brief The index of the alignment matrix row.
+    index_t row_index{};
+    //!\brief The optimal score whose initialisation is implementation defined.
+    score_t score = IMPLEMENTATION_DEFINED;
+
+    /*!\brief Compares the score with the given score and updates the optimum if the new score is bigger than
+     *        the current one.
+     *
+     * \tparam column_index_t The index type for the column index; must model std::unsigned_integral.
+     * \tparam row_index_t The index type for the row index; must model std::unsigned_integral.
+     *
+     * \param[in] compare_score The new score to compare with.
+     * \param[in] column_index The respective column index of the alignment matrix.
+     * \param[in] row_index The respective row index of the alignment matrix.
+     *
+     * \details
+     *
+     * Only updates the current optimum if the new score is greater than the current one. Note in the case of computing
+     * a vectorised alignment only the positions of the simd vector are updated whose score is greater than the current
+     * scores.
+     */
+    template <typename column_index_t, typename row_index_t>
+    void update_if_new_optimal_score(score_t const & compare_score,
+                                     column_index_type<column_index_t> column_index,
+                                     row_index_type<row_index_t> row_index) noexcept;
+}
+#endif //SEQAN3_DOXYGEN_ONLY(1): This code block is only dis
+;
+
+//!\cond
+template <arithmetic score_t>
+struct alignment_optimum<score_t>
+{
+    size_t column_index{};
+    size_t row_index{};
     score_t score{std::numeric_limits<score_t>::lowest()};
-    //!\brief The corresponding coordinate within the alignment matrix.
-    alignment_coordinate coordinate{};
+
+    template <std::integral column_index_t, std::integral row_index_t>
+    constexpr void update_if_new_optimal_score(score_t const & compare_score,
+                                               column_index_type<column_index_t> column_index,
+                                               row_index_type<row_index_t> row_index) noexcept
+    {
+        score = (compare_score > score)
+              ? (this->column_index = column_index.get(), this->row_index = row_index.get(), compare_score)
+              : score;
+    }
 };
 
+template <simd_concept score_t>
+struct alignment_optimum<score_t>
+{
+    using scalar_t = typename simd_traits<score_t>::scalar_type;
+
+    score_t column_index{};
+    score_t row_index{};
+    score_t score{simd::fill<score_t>(std::numeric_limits<scalar_t>::lowest())};
+
+    template <std::integral column_index_t, std::integral row_index_t>
+    constexpr void update_if_new_optimal_score(score_t const & compare_score,
```

Comment:
why constexpr? Will this ever be evaluated in a constexpr context? (And please don't tell me you are preparing compile time alignments :smile: )

Reply:
> And please don't tell me you are preparing compile time alignments 😄 

Of course I thought about it because there are constexpr vector types in the future. 🤓 
But irrespective of that, all functions in the alignments are constexpr (or at least ought to be). It doesn't hurt and there is no reason not to do it.